pydatastructs/linear_data_structures/linked_lists.py

```diff
@@ -0,0 +1,311 @@
+from __future__ import print_function, division
+from pydatastructs.utils.misc_util import _check_type, NoneType
+
+__author__ = 'rohansingh9001'
+
+__all__ = [
+    'DoublyLinkedList',
+    'LinkedList'
+]
+
+# A linked list node 
+class Node: 
+
+    '''
+    Node class for Linked List and Doubly Linked List [ Intended for internal use and not to be imported]
+
+    Parameters
+    ==========
+    
+    For Doubly Linked List use Default constructor(__init__):
+
+        data: type
+            A valid object type.
+            Should be convertible to string using str() method to use print() method on instance
+
+    For Single Linked List use Alternative constructor(singleLink):
+        data: type
+            A valid object type
+            Should be convertible to string using str() method to use print() method on instance
+    
+    Note
+    ====
+
+    classmethod singleLink has been used for Node class for Single linked list due to non existence of a 
+    previous link between the nodes.
+    '''
+
+    __slots__ = ['data', 'next', 'prev']
+    
+    # Constructor to create a new node 
+    def __init__(self, data): 
+        self.data = data 
+        self.next = NoneType
+        self.prev = NoneType
+    #Alternative constructor for Single Linked List
+    @classmethod
+    def singleLink(obj, data):
```

Comment:
The second method is supposed to be an alternative constructor for the Node class when applied to Single LInked Lists. In that case single LInk will be used.

Reply:
Let's analyse the situation.
If we use two separate classes for `Node` one for Doubly Linked lists and other for singly linked lists, then it will lead to unnecessary code.
However, if we use `Node` for Doubly Linked Lists to support the singly linked lists then it will lead to wastage of space.
There can be multiple links in a single node like more than 2, in that case both of the above solutions will become incompatible. 
So, what about passing an extra parameter in the constructor specifying the number of links it will have. In fact, we can take input from the user, the name of each link and create the attributes dynamically, something like in this answer https://stackoverflow.com/a/1325768
I will think more about it's feasibility.